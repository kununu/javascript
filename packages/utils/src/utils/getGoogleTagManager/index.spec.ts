import {getGoogleTagManager, getGoogleTagManagerNoScript} from '.';

describe('GTM function', () => {
  const googleTagManagerId = 'GTM-XXXXX';

  it('should return GTM function without errors', () => {
    const gtm = getGoogleTagManager(googleTagManagerId);

    expect(gtm).toMatchSnapshot();
  });
  it('should return GTM no-script frame without errors', () => {
    const gtm = getGoogleTagManagerNoScript(googleTagManagerId);

    expect(gtm).toMatchSnapshot();
  });
});
