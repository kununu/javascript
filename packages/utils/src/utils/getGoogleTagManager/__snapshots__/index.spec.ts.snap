// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GTM function should return GTM function without errors 1`] = `
"// google tag manager
    (function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
    new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
    j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
    'https://www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);
    })(window,document,'script','dataLayer', 'GTM-XXXXX');"
`;

exports[`GTM function should return GTM no-script frame without errors 1`] = `"<iframe src=\\"https://www.googletagmanager.com/ns.html?id=GTM-XXXXX\\" height=\\"0\\" width=\\"0\\" style=\\"display:none;visibility:hidden\\"></iframe>"`;
